experiment-1-typical.ipynb

```diff
@@ -474,80 +469,7 @@
-    "reference_results = reference_results.reset_index()\n",
-    "\n",
-    "filtered_results = results_df\n",
-    "filtered_results = filtered_results[filtered_results.database_path == 'metabolomics/db/mol_db1.pickle']\n",
-    "#filtered_results = filtered_results[filtered_results.fdr <= 0.5]\n",
-    "filtered_results = filtered_results[filtered_results.adduct != '']\n",
-    "filtered_results = filtered_results[filtered_results.modifier == '']\n",
-    "filtered_results = filtered_results.rename({'mol': 'formula'}, axis=1)\n",
-    "filtered_results = filtered_results[['chaos', 'spatial', 'spectral', 'msm', 'fdr', 'formula', 'adduct', 'modifier']]\n",
-    "\n",
-    "merged_results = reference_results.merge(filtered_results, how='outer', left_on=['formula','adduct'], right_on=['formula','adduct'], suffixes=['_ref',''])\n",
-    "merged_results = merged_results.sort_values(['formula','adduct'])"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "common_results[['spatial', 'spatial_ref']]"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "common_results = merged_results[merged_results.fdr.notna() & merged_results.fdr_ref.notna()]\n",
-    "spatial_wrong = common_results[np.abs((common_results.spatial - common_results.spatial_ref)) >= 0.01]\n",
-    "assert len(spatial_wrong) < 5, spatial_wrong # Higher tolerance because older DSs seem to have used a different algorithm\n",
-    "spectral_wrong = common_results[np.abs((common_results.spectral - common_results.spectral_ref)) >= 0.001]\n",
-    "assert len(spectral_wrong) <= 1, spectral_wrong\n",
-    "chaos_wrong = common_results[np.abs((common_results.chaos - common_results.chaos_ref)) >= 0.01]\n",
-    "assert len(chaos_wrong) <= 1, chaos_wrong\n",
-    "msm_wrong = common_results[np.abs((common_results.msm - common_results.msm_ref)) >= 0.001]\n",
-    "assert len(msm_wrong) <= 10, msm_wrong"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "\n",
-    "def quantize_fdr(fdr):\n",
-    "    if fdr <= 0.05: return 1\n",
-    "    if fdr <= 0.1: return 2\n",
-    "    if fdr <= 0.2: return 3\n",
-    "    if fdr <= 0.5: return 4\n",
-    "    return 5\n",
-    "    \n",
-    "fdr_level = merged_results.fdr.apply(quantize_fdr)\n",
-    "fdr_ref_level = merged_results.fdr_ref.apply(quantize_fdr)\n",
-    "\n",
-    "fdr_exact = merged_results[fdr_level == fdr_ref_level]\n",
-    "fdr_near = merged_results[np.abs(fdr_level - fdr_ref_level) <= 1]\n",
-    "print(len(fdr_exact), len(fdr_near), len(merged_results))\n",
-    "assert len(fdr_exact) >= len(merged_results) * 0.5\n",
-    "assert len(fdr_near) >= len(merged_results) * 0.8"
+    "checked_results = pipeline.check_results()"
```

Comment:
I ran the project with the "big" dataset and got `annotation-pipeline: 5 annotations with incorrect spectral metric`, can you detail what do you expect to get at this part? what is the "good range" for you?

Reply:
I've pushed a change that improves the output:
```
2019-07-02 12:33:10,542 [INFO] annotation-pipeline: Missing annotations: 0 (PASS)
2019-07-02 12:33:10,543 [INFO] annotation-pipeline: Incorrect spatial metric: 0 (PASS)
2019-07-02 12:33:10,543 [INFO] annotation-pipeline: Incorrect spectral metric: 5 (PASS)
2019-07-02 12:33:10,543 [INFO] annotation-pipeline: Incorrect chaos metric: 0 (PASS)
2019-07-02 12:33:10,544 [INFO] annotation-pipeline: Incorrect MSM: 0 (PASS)
2019-07-02 12:33:10,544 [INFO] annotation-pipeline: FDR changed: 54 (PASS)
2019-07-02 12:33:10,544 [INFO] annotation-pipeline: FDR changed significantly: 0 (PASS)
2019-07-02 12:33:10,544 [INFO] annotation-pipeline: All checks pass
```